Mathlib/Data/Finsupp/Defs.lean

```diff
@@ -714,6 +714,10 @@ theorem mapRange_comp (f : N → P) (hf : f 0 = 0) (f₂ : M → N) (hf₂ : f
     (g : α →₀ M) : mapRange (f ∘ f₂) h g = mapRange f hf (mapRange f₂ hf₂ g) :=
   ext fun _ => rfl
 
+@[simp]
+lemma mapRange_mapRange (e₁ : N → P) (e₂ : M → N) (he₁ he₂) (f : α →₀ M) :
```

Comment:
I have no idea what the preferred direction would be for this `simp` lemma. Any particular reason it goes from two function applications to a function composition?

Reply:
The other direction wouldn't work as you can't deduce `e₁ 0 = 0` and `e₂ 0 = 0` from `(e₁ ∘ e₂) 0 = 0`, while you can deduce `(e₁ ∘ e₂) 0 = 0` from `e₁ 0 = 0` and `e₂ 0 = 0`

Note that the other direction is given by `Finsupp.mapRange_comp` just above

Reply:
Ok, works for me!